lightning/src/ln/chan_utils.rs

```diff
@@ -550,6 +554,292 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransactionInfo)
+pub struct ChannelStaticInfo {
+	/// Broadcaster public keys
+	pub broadcaster_pubkeys: ChannelPublicKeys,
+	/// Counter-signatory public keys
+	pub countersignatory_pubkeys: ChannelPublicKeys,
+	/// The funding outpoint
+	pub funding_outpoint: OutPoint,
+	/// The contest delay selected by the countersignatory
+	pub contest_delay: u16,
+	/// Whether the channel is outbound from the point of view of the broadcaster
+	pub is_outbound: bool,
+}
+
+/// This class tracks the information needed to build a holder's commitment transaction and to actually
+/// build and sign.  It is used for holder transactions that we sign only when needed.
+pub struct HolderCommitmentTransactionInfo {
+	/// The non-party-specific transaction information
+	pub info: CommitmentTransactionInfo,
+	/// Our counterparty's signature for the transaction
+	pub counterparty_sig: Signature,
+	/// All HTLCs (including dust ones) with an optional counterparty signature
+	pub htlcs_with_sig: Vec<(HTLCOutputInCommitment, Option<Signature>)>,
+}
+
+impl HolderCommitmentTransactionInfo {
+	pub(crate) fn to_holder_commitment_tx<T: secp256k1::Signing + secp256k1::Verification>(&self, channel_static_info: &ChannelStaticInfo, secp_ctx: &Secp256k1<T>) -> HolderCommitmentTransaction {
+		self.info.to_holder_commitment_tx(self.counterparty_sig, self.htlcs_with_sig.clone(), channel_static_info, secp_ctx)
+	}
+}
+
+#[derive(Clone)]
+/// This class tracks the information needed to build a commitment transaction and to actually
+/// build and sign.  It is used for holder transactions that we sign only when needed
+/// and for transactions we sign for the counterparty.
+///
+/// This class can be used inside a signer implementation to generate a signature given the relevant
+/// secret key.
+pub struct CommitmentTransactionInfo {
```

Comment:
looking for a better name for this

Reply:
I also have a particular distain for names ending with suffixes like "Info" or "Data. :) They don't tend to add any clarity.

Seems like this is essentially the builder pattern. How about `CommitmentTransactionBuilder`? Would be great if we could formulate the related interfaces in such a manner, if this makes sense, rather than using methods with long parameter lists like `new_with_auxiliary_htlc_data`. Then have `build` construct everything as needed.

Reply:
`*Builder` was also my first thought too, but it would then be strange that this would be persisted in `OnchainTxHandler.holder_commitment` once this PR is fleshed out.  i.e. builders are usually ephemeral.

Reply:
How do you feel about `CommitmentState`?  This basically encapsulates everything that changes about the commitment transaction (and attached HTLCs) as it changes state through the channel lifetime.

Cc @ariard